app/views/games/index.html.erb

```diff
@@ -7,17 +7,21 @@
       <th></th>
       <th></th>
       <th></th>
+      <th></th>
     </tr>
   </thead>
 
   <tbody>
     <% @games.each do |game| %>
-      <tr>
-        <td><%= game.name %></td>
-        <td><%= link_to 'Show', game %></td>
-        <td><%= link_to 'Edit', edit_game_path(game) %></td>
-        <td><%= link_to 'Destroy', game, method: :delete, data: { confirm: 'Are you sure?' } %></td>
-      </tr>
+      if game.available? do
+        <tr>
+          <td><%= game.name %></td>
+          <td><%= link_to 'Join', games, method: :update %></td>
```

Comment:
Looking at this I have an idea :bulb: :)

For this button, there is a way to pass query parameters along as well.

See if you can update it to send `game: { black_player_id: current_user.id }` to the controller.

Then you wouldn't need to change the controller at all (could keep the existing code that was there before)